### PR TITLE
[viessmann] add status channel for OneTimeCharge

### PR DESCRIPTION
**Pull request by rogrun**
This PR: 
Adds a channel which provides the state of OneTimeCharge. (Device Thing)
Adds the ability to disable polling. (Bridge Thing)
Adds channels for manual polling. (Bridge Thing)

Improves the channel type for channels w/o commands to read-only.

Also fixes some typo.

Signed-off-by: Ronny Grun <ronny.grun@t-online.de>

### CODE REVIEW

**Comment by rogrun** [comment]
Ready for Review.

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/config/BridgeConfiguration.java [hunk 1]
```diff
@@ -31,4 +31,5 @@ public class BridgeConfiguration {
     public int bufferApiCommands = 450;
     public int pollingInterval = 0;
     public int pollingIntervalErrors = 60;
+    public Boolean disablePolling = false;
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/config/BridgeConfiguration.java]:
```suggestion
    public boolean disablePolling = false;
```

Use primitives where possible.

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java [hunk 2]
```diff
@@ -200,10 +218,13 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                                 : (ViessmannBridgeHandler) bridge.getHandler();
                         if (bridgeHandler != null) {
                             if (!bridgeHandler.setData(uri, param) || initState) {
-                                initChannelState();
+                                ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+                                executorService.schedule(this::initChannelState, initStateDelay, TimeUnit.SECONDS);
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java]: Is there a reason why you don't use the integrated `scheduler`?

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java [hunk 3]
```diff
@@ -132,7 +132,24 @@ private boolean errorChannelsLinked() {
...
-        // Nothing to handle here currently
+        if (channelUID.getId().equals(CHANNEL_RUN_QUERY_ONCE)) {
+            if (command instanceof OnOffType) {
+                if (command == OnOffType.ON) {
+                    logger.debug("Received command: CHANNEL_RUN_QUERY_ONCE");
+                    pollingFeatures();
+                    updateState(CHANNEL_RUN_QUERY_ONCE, OnOffType.OFF);
+                }
+            }
+        }
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java]:
```suggestion
        if (channelUID.getId().equals(CHANNEL_RUN_QUERY_ONCE) && OnOffType.ON.equals(command)) {
            logger.debug("Received command: CHANNEL_RUN_QUERY_ONCE");
            pollingFeatures();
            updateState(CHANNEL_RUN_QUERY_ONCE, OnOffType.OFF);
        }
```

also below

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java [hunk 4]
```diff
@@ -162,7 +179,7 @@ public void initialize() {
             setConfigInstallationGatewayId();
         }
 
-        if (errorChannelsLinked()) {
+        if (errorChannelsLinked() && !config.disablePolling) {
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java]:
```suggestion
        if (!config.disablePolling && errorChannelsLinked()) {
```

Might not be the biggest improvement, but the `!config.disablePolling` check is cheap compared to the `isLinked` check, so it's better to do it first.

**Review by J-N-K — CHANGES_REQUESTED** [review]
Thanks for all the clean-up. I only left some minor comments.

**Review by J-N-K — DISMISSED** [review]

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java (hunk 2 above)
Comment (rogrun) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java]: I don't know the integrated scheduler. I changed it. Thanks!

**Review by J-N-K — APPROVED** [review]

**Pull request merged**